src/index.js

```diff
@@ -48,6 +48,7 @@ const App = () => {
 
         setTodoInputValue('');
         setDateInputValue('');  
+        setHasTaskInputBeenTouched(false);
```

Comment:
This means `hasTaskInputBeenTouched` is not changed on the fly...

Reply:
@KevinHong913 
 
Problem: `onBlur={() => setHasTaskInputBeenTouched(true)}` sets a boolean, and once it has been set to true, it needs to be set back to false. Or else `setHasTaskInputBeenTouched` will always be true causing the invalid class to be true always. And I only want to reset this only at `handleAddItem`, because `handleClearItems` is removing a task and they are 2 separate actions. 

1. Looked into form resetting, most answers are to create an empty state that captures all the fields that need to be emptied, and call and apply this anywhere you need. This is achievable but it doesn't really solve the manually setting it back to false problem.

2. Use a react form library, like react-hook-form.

Reply:
Is it causing any problem now? You're logic seems fine me.